core/includes/form.inc

```diff
@@ -3023,7 +3023,7 @@ function form_process_password_confirm($element) {
     '#title' => t('New password'),
     '#value' => empty($element['#value']) ? NULL : $element['#value']['pass1'],
     '#required' => $element['#required'],
-    '#attributes' => array('class' => array('password-field')),
+    '#attributes' => $element['#attributes'] + array('class' => array('password-field')),
```

Comment:
I'll be inserting comments to explain my changes:

Here, I'm merging any preset key/value pairs in `$elements['#attributes]` sent by the form function, with the password-field class. 

The way this process function works is the following:
1. The form function sends a single subarray with the form `password_confirm` element, as in:

```php
  $form['password'] = array(
    '#type' => 'password_confirm',
    '#title' => 'Password',
    '#maxlength' => 64,
    '#size' => 15,
    // '#disabled' => TRUE,
    '#attributes' => array(
      'spellcheck' => 'true',
    ),
  );
```
2. This process form splits the parent element into two `password` subelements, one for the actual password, and one for the confirmed password. 

So this line merges any preset attributes from the parent form element into the two subelements

The `+` sign is a special array operator to merge the two arrays.